lib/create_household/view/create_household.dart

```diff
@@ -0,0 +1,200 @@
+                  context: context,
+                  barrierDismissible: false,
+                  builder: (_) => AlertDialog(
+                        title: const Text('Are you sure?'),
+                        content:
+                            const Text('Any unsaved changes will be lost.'),
+                        actions: [
+                          ElevatedButton(
+                              onPressed: () {
+                                willLeave = true;
+                                Navigator.of(context).pop();
+                              },
+                              child: const Text('Yes')),
+                          TextButton(
+                              onPressed: () => Navigator.of(context).pop(),
+                              child: const Text('No'))
+                        ],
+                      ));
+              return willLeave;
+            },
+            child: Scaffold(
+              appBar: AppBar(
+                title: const Text('Create new household'),
+                actions: [
+                  IconButton(
+                      onPressed: () => {
+                            if (formKey.currentState!.saveAndValidate())
+                              {
+                                context.read<CreateHouseholdBloc>().add(
+                                    NewHouseholdSaveRequested(
+                                        household: Household(
+                                            householdId: formKey.currentState!
+                                                .value['householdId'],
+                                            sensitizationDate: formKey
+                                                .currentState!
+                                                .value['sensitizationDate'],
+                                            geoLocation: formKey.currentState!
+                                                .value['geoLocation'],
+                                            comments: formKey.currentState!
+                                                    .value['comments'] ??
+                                                ""))),
+                                Navigator.pop(context),
+                              }
+                          },
+                      icon: const Icon(Icons.save))
+                ],
+              ),
+              body: Padding(
+                padding: const EdgeInsets.all(8.0),
+                child: FormBuilder(
+                  key: formKey,
+                  child: Column(children: [
+                    FormBuilderTextField(
+                      name: 'householdId',
+                      decoration: const InputDecoration(
+                        label: Text('Household ID'),
+                        icon: Icon(Icons.house),
+                      ),
+                      onChanged: (value) => changed = true,
+                      validator: FormBuilderValidators.compose([
```

Comment:
could be nice to extract the validators into separate class

Reply:
Will leave this for now